File: backend/app/api/v1/endpoints/ai.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session
from app.api.v1.deps import get_db, require_office_or_supervisor_or_admin as require_power
from app.ai.service import recommend_replenish, recommend_transfer, detect_anomalies, RecoParams

router = APIRouter(prefix="/ai", tags=["ai"])

@router.get("/reco/replenish")
def ai_reco_replenish(days: int = Query(28, ge=7, le=120), leadtime: int = Query(3, ge=0, le=30),
                      target_days: int = Query(7, ge=1, le=60), min_qty: int = Query(1, ge=0, le=100),
                      same_city_only: bool = Query(True), same_network_only: bool = Query(True),
                      db: Session = Depends(get_db), _: None = Depends(require_power)):
    p = RecoParams(days=days, leadtime=leadtime, target_days=target_days, min_qty=min_qty,
                   same_city_only=same_city_only, same_network_only=same_network_only)
    return recommend_replenish(db, p)

@router.get("/reco/transfer")
def ai_reco_transfer(days: int = Query(28, ge=7, le=120), leadtime: int = Query(3, ge=0, le=30),
                     target_days: int = Query(7, ge=1, le=60), min_qty: int = Query(1, ge=0, le=100),
                     same_city_only: bool = Query(True), same_network_only: bool = Query(True),
                     db: Session = Depends(get_db), _: None = Depends(require_power)):
    p = RecoParams(days=days, leadtime=leadtime, target_days=target_days, min_qty=min_qty,
                   same_city_only=same_city_only, same_network_only=same_network_only)
    return recommend_transfer(db, p)

@router.get("/alerts/anomalies")
def ai_alerts_anomalies(db: Session = Depends(get_db), _: None = Depends(require_power)):
    return detect_anomalies(db)

File: frontend/src/shared/api/ai.ts
import { api } from './http';

export async function fetchReplenish(params: {days?: number; leadtime?: number; target_days?: number; min_qty?: number; same_city_only?: boolean; same_network_only?: boolean} = {}) {
  const { data } = await api.get('/api/v1/ai/reco/replenish', { params });
  return data as { items: any[] };
}

export async function fetchTransfer(params: {days?: number; leadtime?: number; target_days?: number; min_qty?: number; same_city_only?: boolean; same_network_only?: boolean} = {}) {
  const { data } = await api.get('/api/v1/ai/reco/transfer', { params });
  return data as { pairs: any[] };
}

export async function fetchAnomalies() {
  const { data } = await api.get('/api/v1/ai/alerts/anomalies');
  return data as { items: any[] };
}
